logstash-core/lib/logstash/output_delegator_strategies/legacy_worker.rb

```diff
@@ -0,0 +1,27 @@
+# Remove this in Logstash 6.0
+module LogStash module OutputDelegatorStrategies class LegacyWorker
+  ::LogStash::OutputDelegatorStrategyRegistry.instance.register(:legacy_worker, self)
```

Comment:
I am concerned about this register statement.

In practice this is not a problem because we are in full control of how and when these classes are retrieved. That said, the current code would give the casual reader the idea that this is OK but there is a bug lurking in it. The class is registered without methods - they are added shortly after registration. However, if the class is retrieved and instantiated before the methods are added the instance will have no public API.

Consider moving the registration to after the methods are added. Alternatives:
```ruby
module LogStash
  module OutputDelegatorStrategies class LegacyWorker
    # rest of class definition omitted for illustration
  end; end

  OutputDelegatorStrategyRegistry.instance.register(
    :legacy_worker, OutputDelegatorStrategies::LegacyWorker)
end
```
or
```ruby
module LogStash module OutputDelegatorStrategies class LegacyWorker
  # rest of class definition omitted for illustration
  
  ::LogStash::OutputDelegatorStrategyRegistry.instance.register(:legacy_worker, self)
end; end; end
```